docs/guide/interactions/slash_commands.rst

```diff
@@ -0,0 +1,1519 @@
+
+typing.Optional
+++++++++++++++++
+
+Discord supports optional parameters, wherein a user doesn't need to provide a value during invocation.
+
+A parameter is considered optional if its assigned a default value and/or annotated
+to :obj:`~typing.Optional`.
+
+For example, this command displays a given user's avatar, or the current user's avatar:
+
+.. code-block:: python
+
+    from typing import Optional
+
+    @client.tree.command()
+    async def avatar(interaction: discord.Interaction, user: Optional[discord.User] = None):
+        avatar = (user or interaction.user).display_avatar
+        await interaction.response.send_message(avatar.url)
+
+On Discord:
+
+.. image:: /images/guide/app_commands/avatar_command_optional_preview.png
+    :width: 300
+
+`Python version 3.10+ union types <https://peps.python.org/pep-0604/>`_ are also supported instead of :obj:`typing.Optional`.
+
+typing.Union
++++++++++++++
+
+Some types comprise of multiple other types. For example, a ``MENTIONABLE`` type parameter can point to any of these:
+
+- :class:`discord.User`
+- :class:`discord.Member`
+- :class:`discord.Role`
+
+To specify in code, a parameter should annotate to a :obj:`typing.Union` with all the different models:
+
+.. code-block:: python
+
+    from typing import Union
+
+    @client.tree.command()
+    async def something(
+        interaction: discord.Interaction,
+        mentionable: Union[discord.User, discord.Member, discord.Role]
+    ):
+        await interaction.response.send_message(
+            f'i got: {mentionable}, of type: {mentionable.__class__.__name__}'
+        )
+
+Types that point to other types also don't have to include everything.
+For example, a ``CHANNEL`` type parameter can point to any channel in a guild,
+but can be narrowed down to a specific set of channels:
+
+.. code-block:: python
+
+    from typing import Union
+
+    @client.tree.command()
+    async def channel_info(interaction: discord.Interaction, channel: discord.abc.GuildChannel):
+        # Everything except threads
+        pass
+
+    @client.tree.command()
+    async def channel_info(interaction: discord.Interaction, channel: discord.TextChannel):
+        # Only text channels
+        pass
+
+    @client.tree.command()
+    async def channel_info(interaction: discord.Interaction, channel: Union[discord.Thread, discord.VoiceChannel]):
+        # Threads and voice channels only
+        pass
+
+.. note::
+
+    Union types can't mix Discord types.
+
+    Something like ``Union[discord.Member, discord.TextChannel]`` isn't possible.
+
+Refer to the :ref:`type conversion table <type_conversion>` for full information on sub-types.
+
+Describing
++++++++++++
+
+Descriptions are added to parameters using the :func:`.app_commands.describe` decorator,
+where each keyword is treated as a parameter name.
+
+.. code-block:: python
+
+    @client.tree.command()
+    @app_commands.describe(
+        liquid="what type of liquid is on the wall",
```

Comment:
Noticed that the param descriptions were not capitalized, since the ones in the examples are. Also missing the command description from what I can tell.

```suggestion
        liquid="What type of liquid is on the wall",
        amount="How much of it is on the wall"
```

Reply:
Yea this makes total sense